src/components/Articles/SingleArticle.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, useContext } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "../../api";
+import { Card } from "@mui/material";
+import { ThemeContext } from "../../context/Theme";
+
+export const SingleArticle = () => {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const { theme } = useContext(ThemeContext);
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((article) => setArticle(article))
+      .catch((err) => console.log(err));
+  }, [article_id]);
+
+  if (!article) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div className={`single-article ${theme}`}>
+      <Card
```

Comment:
Great to see you have implemented context for a light/dark toggle mode for your app!